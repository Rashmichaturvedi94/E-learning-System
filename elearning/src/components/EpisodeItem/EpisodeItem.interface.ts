export interface EpisodeItemProps {
  item?: any;
}
