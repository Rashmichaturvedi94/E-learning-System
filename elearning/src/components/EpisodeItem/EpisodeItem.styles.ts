import styled from 'styled-components/native';
import { TouchableOpacity, View, Image, Text } from 'react-native';

export const TouchPlay = styled(TouchableOpacity)`
  height: 50px;
  width: 50px;
  justify-content: center;
  align-items: center;
`;

export const ListItemContainer = styled(View)`
  margin-left: 20px;
  flex: 1;
  align-items: flex-start;
`;

export const ListTitle = styled(Text)`
  font-weight: bold;
`;

export const ListImage = styled(Image)`
  width: 40px;
  height: 40px;
`;

export const ListView = styled(View)`
  flex: 3;
  background-color: white;
`;

export const ListItemView = styled(View)`
  flex-direction: row;
  flex: 1;
  margin: 0px 16px;
`;
