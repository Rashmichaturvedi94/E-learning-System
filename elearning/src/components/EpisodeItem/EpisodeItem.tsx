import React, { FC } from 'react';
import { Text } from 'react-native';
import { Icon } from 'react-native-elements';
import {
  ListItemView,
  ListImage,
  ListItemContainer,
  ListTitle,
  TouchPlay,
} from './EpisodeItem.styles';
import { EpisodeItemProps } from './EpisodeItem.interface';

export const EpisodeItem: FC<EpisodeItemProps> = ({ item }) => {
  return (
    <ListItemView>
      <ListImage source={{ uri: item.url }} />
      <ListItemContainer>
        <ListTitle>{item.title}</ListTitle>
        <Text>{item.desc}</Text>
      </ListItemContainer>
      <TouchPlay>
        <Icon name="info" />
      </TouchPlay>
    </ListItemView>
  );
};
