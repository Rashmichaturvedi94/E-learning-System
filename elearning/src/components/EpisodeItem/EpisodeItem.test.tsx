import React from 'react';
import { render } from '@testing-library/react-native';
import { EpisodeItem } from './EpisodeItem';

const episode = {
  title: 'test',
  desc: 'testdec',
  url: 'https://upload.wikimedia.org/wikipedia/commons/7/78/Swift_logo.png',
};

describe('components/EpisodeItem', () => {
  it('should render', () => {
    const component = render(<EpisodeItem item={episode} />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
