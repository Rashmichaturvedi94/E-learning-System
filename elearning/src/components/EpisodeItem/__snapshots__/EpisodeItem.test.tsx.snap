// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/EpisodeItem should render 1`] = `
<View
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexDirection": "row",
        "flexGrow": 1,
        "flexShrink": 1,
        "marginBottom": 0,
        "marginLeft": 16,
        "marginRight": 16,
        "marginTop": 0,
      },
    ]
  }
>
  <Image
    source={
      Object {
        "uri": "https://upload.wikimedia.org/wikipedia/commons/7/78/Swift_logo.png",
      }
    }
    style={
      Array [
        Object {
          "height": 40,
          "width": 40,
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "alignItems": "flex-start",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "marginLeft": 20,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontWeight": "bold",
          },
        ]
      }
    >
      test
    </Text>
    <Text>
      testdec
    </Text>
  </View>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "height": 50,
        "justifyContent": "center",
        "opacity": 1,
        "width": 50,
      }
    }
  >
    <View
      style={
        Object {
          "overflow": "hidden",
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "justifyContent": "center",
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#242424",
                  "fontSize": 24,
                },
                Object {
                  "backgroundColor": "transparent",
                },
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
            testID="iconIcon"
          >
            î¢Ž
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
